src/GraphQL.Tests/Bugs/ScalarsInputTest.cs

```diff
@@ -172,21 +180,40 @@ public void Should_Accept_Long_As_Long_In_Literal()
             AssertQuerySuccess(query, expected, null, null);
         }
 
+        [Fact]
+        public void Should_Not_Accept_String_As_Int_In_Literal()
+        {
+            var query = @"mutation { int(number: ""100"") }";
+            string expected = null;
+            var result = AssertQueryWithErrors(query, expected, expectedErrorCount: 1, executed: false);
+            result.Errors[0].Message.ShouldBe("Argument \"number\" has invalid value \"100\".\nExpected type \"Int\", found \"100\".");
+        }
+
+        [Fact]
+        public void Should_Not_Accept_String_As_Int_In_Variable()
+        {
+            var query = @"mutation AAA($val : Int!) { int(number: $val) }";
+            var expected = @"{ ""int"": 100 }";
+            var result = AssertQueryWithErrors(query, expected, inputs: @"{ ""val"": ""100"" }".ToInputs(), expectedErrorCount: 1, executed: false);
+            result.Errors[0].Message.ShouldBe(@"Variable '$val' is invalid. Unable to convert '100' to 'Int'");
+        }
+
         [Fact]
         public void Should_Not_Accept_String_As_Long_In_Literal()
         {
             var query = @"mutation { long(number: ""100"") }";
             string expected = null;
-            AssertQueryWithErrors(query, expected, expectedErrorCount: 1, executed: false);
+            var result = AssertQueryWithErrors(query, expected, expectedErrorCount: 1, executed: false);
+            result.Errors[0].Message.ShouldBe("Argument \"number\" has invalid value \"100\".\nExpected type \"Long\", found \"100\".");
```

Comment:
@Shane32 Also I would like to change all `"` to `'` from error messages in separate PR.